src/dolphin/atmosphere/troposphere.py

```diff
@@ -165,6 +168,16 @@ def estimate_tropospheric_delay(
             logger.warning(f"Weather-model files do not exist for {date_str}, skipping")
             continue
 
+        secondary_time = oput.get_zero_doppler_time(slc_files[secondary_date][0])
+        if len(slc_files[reference_date]) == 0:
+            # this is for when we have compressed slcs but the actual
+            # reference date does not exist in the input data
+            reference_time = datetime.datetime.combine(
+                reference_date[0].date(), secondary_time.time()
+            )
```

Comment:
Ah ok I see... so if we had longer filenames like `%Y%m%dT%H%M%S` then this line wouldn't be needed, but you're combining the date-only ref date with the full datetime secondary time?

looks like a good immediate fix, and i'm not sure this alone outweighs the annoyance of having every filename by doubled in length